### DISCUSSION TITLE
Is it possible to build for debug, watching the files?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#13153] [General - Components / Directives / etc]
- I know how to develop an app using `quasar dev`
- I know how to build an app for production via `quasar build` (or `quasar build --debug`)

I have the following problem: I develop a backend that will serve the Quasar front (either embed it in the binary (Golang), or serving the `spa` directory with the build.

When I develop both in parallel, I have the backend exposing the API on `http://localhost:XX`, and Quasar on `http://localhost:8080`. It works great. I then deploy and there are some problems I would like to debug, but even with the debug flag it is strange.

So what I would like to do is to have **Quasar continuously build on disk instead of serving from memory** (`quasar dev --writeOnDisk` kind of command) so that I can serve that from my backend, as if it was prod.

The workaround is to build myself at changes, or use a building pipe (I used to use that some years ago with first builds). But if this is part of Quasar why reinvent the wheel :)

**Comment by bklik** [comment]
I think this is related to #10180 

I too came here looking for this solutions. Something similar to how Angular does it with `--watch':
https://angular.io/cli/build

If it hasn't been made clear, we're all looking for a way to constantly be developing against production, rather than the dev instance.

**Comment by MilosPaunovic** [comment]
Add `--debug` flag.

**Reply by wsw70** [reply]
> Add `--debug` flag.

This is still going to start a dev instance, right? Not build final files for deployment?

**Reply by MilosPaunovic** [reply]
`quasar build --debug` will build files for debugging purposes.

**Reply by bklik** [reply]
Thank you, but that does not scratch the itch.

As mentioned, the desire is to run a `quasar build --watch` command. It watches for development changes, and then rebuilds production every time a file is changed (like Angular's `ng build --watch` command).

**Comment by pdanpdan** [comment]
Why would you need this?

**Reply by bklik** [reply]
Sometimes bugs surface in production that you don't see in the development server. Or, you want you and your team developing and testing against something that is always as close as it can be to production.

Other people have mentioned on other similar request (https://github.com/vuejs/vue-cli/issues/1317) that they have a project that is integrating with another project, so the dev server won't work.

**Reply by benzkji** [reply]
also, if you develop against a backend running for example in Django, it will be running it's own dev server, on another port. instead of configuring and messing with CORS on your local machine, with `quasar dev --writeToDisk` you can/could serve everything with one webserver.

Alternativly, one could proxy everything through webpack/vite (dont know if vite does this?).
